sklearn/metrics/_ranking.py

```diff
@@ -1701,7 +1701,7 @@ def top_k_accuracy_score(y_true, y_score, *, k=2, normalize=True,
     if y_type == 'binary':
         if k == 1:
             threshold = .5 if y_score.min() >= 0 and y_score.max() <= 1 else 0
-            y_pred = (y_score > threshold).astype(np.int)
+            y_pred = (y_score > threshold).astype(int)
```

Comment:
By default, I think we should use `np.int64` whenever we used to use `np.int` (which is the implicit conversion that happens on 64 bit linux) unless we know that this are mathematically constrained to be positive quantities, in which case we might prefer to use `np.uint64`.

Reply:
I agree with you that we should use fixed-size aliases to avoid platform-dependent definitions. I use 'find . -name "*.py" -print0 | xargs -0 sed -i "s/\bnp.int\b/int/g"` to make the PR but I should have specified a more specific data type in some cases.

Fixing 😉.